pageserver/src/tenant/remote_timeline_client.rs

```diff
@@ -361,11 +365,42 @@ impl RemoteTimelineClient {
         Ok(())
     }
 
+    /// Initialize the queue in stopped state. Used in startup path
+    /// to continue deletion operation interrupted by pageserver crash or restart.
+    pub fn init_upload_queue_stopped_to_continue_deletion(
+        &self,
+        index_part: &IndexPart,
+    ) -> anyhow::Result<()> {
+        {
+            let mut upload_queue = self.upload_queue.lock().unwrap();
+            upload_queue.initialize_with_current_remote_index_part(index_part)?;
+            self.update_remote_physical_size_gauge(Some(index_part));
+        }
```

Comment:
can't this be a parameter to stop method ? or maybe a different function
self.init_upload_queue(index_part) ?

Reply:
oh well, I am thinking about avoiding taking the lock twice, but this isn't on the hot path so I am resolving this one

Reply:
Yeah, agree that it looks ugly, I think https://github.com/neondatabase/neon/issues/4378 is the place for it